base/bucket_n1ql.go

```diff
@@ -14,6 +14,7 @@ const BucketQueryToken = "$_bucket"   // Token used for bucket name replacement
 const MaxQueryRetries = 30            // Maximum query retries on indexer error
 const IndexStateOnline = "online"     // bucket state value, as returned by SELECT FROM system:indexes
 const IndexStateDeferred = "deferred" // bucket state value, as returned by SELECT FROM system:indexes
+const IndexStatePending = "pending"   // bucket state value, as returned by SELECT FROM system:indexes
```

Comment:
Should this be "index state value"?  Also, would it be possible to point to docs that have the full set of states?

Reply:
It might also be worth adding a blurb on what these states mean, eg:

* deferred: the index is being created in the background by Couchbase
* pending: the index is created, but has not been built yet.

Reply:
Adding 'Value' to the constant names just feels wordy, without adding useful information.  

I'll include a bit more info on what the states mean.